File: src/test/java/dev/retrotv/random/RandomStringGeneratorTest.java
package dev.retrotv.random;

import dev.retrotv.random.enums.SecurityStrength;
import org.junit.jupiter.api.Test;

import java.security.SecureRandom;

class RandomStringGeneratorTest {

    @Test
    void test_randomStringGenerator() {
        PasswordGenerator pg = new PasswordGenerator(SecurityStrength.HIGH, new SecureRandom());
        pg.generate(8);
        System.out.println(pg.getValue());
    }
}

File: src/test/kotlin/dev/retrotv/random/RandomStringGeneratorTest.kt
package dev.retrotv.random

import dev.retrotv.random.value.getCapitalLetters
import dev.retrotv.random.value.getNumbers
import dev.retrotv.random.value.getSmallLetters
import dev.retrotv.random.value.getSpecialChars
import kotlin.test.Test

class RandomStringGeneratorTest {

    @Test
    fun test() {
        val rp = UniversalRandomStringGenerator(getSmallLetters(), getNumbers(), getCapitalLetters(), getSpecialChars())
        rp.disableAllCharGroupLeastOne()
        rp.generate(20)
        println(rp.getString())
    }
}